third_party/terraform/utils/provider.go.erb

```diff
@@ -98,12 +97,45 @@ func Provider() terraform.ResourceProvider {
 			"google_project_services":                dataSourceGoogleProjectServices(),
 			"google_service_account":                 dataSourceGoogleServiceAccount(),
 			"google_service_account_key":             dataSourceGoogleServiceAccountKey(),
-			"google_storage_bucket_object":           dataSourceGoogleStorageBucketObject(),
 			"google_storage_object_signed_url":       dataSourceGoogleSignedUrl(),
 			"google_storage_project_service_account": dataSourceGoogleStorageProjectServiceAccount(),
 		},
 
-		ResourcesMap: mergeResourceMaps(
+		ResourcesMap: ResourceMap(),
+
+		ConfigureFunc: providerConfigure,
+	}
+}
+
+func mergeResourceMaps(ms ...map[string]*schema.Resource) (map[string]*schema.Resource, error) {
+	merged := make(map[string]*schema.Resource)
+	duplicates := []string{}
+
+	for _, m := range ms {
+		for k, v := range m {
+			if _, ok := merged[k]; ok {
+				duplicates = append(duplicates, k)
+			}
+
+			merged[k] = v
+		}
+	}
+
+	var err error
+	if len(duplicates) > 0 {
+		err = fmt.Errorf("saw duplicates in mergeResourceMaps: %v", duplicates)
+	}
+
+	return merged, err
+}
+
+func ResourceMap() map[string]*schema.Resource {
```

Comment:
nit: I'd rename to BuildResourceMap or something. Naming a function the same name as the resource it's returning doesn't indicate what it's really doing.

Reply:
If Go didn't require me to break this out into a function, it would totally just be a const - imo `Build` or `Get` or another prefix makes it sound like more is happening than actually is, so calling it the name of the resource is most clear. We're not really building or getting anything, this fn just returns the ResourceMap as a property.

Reply:
What is happening inside that method is a call to another method which is building/merging a resource list. I wouldn't expect a constructor'y looking method to return a fully populated map of merged resources with suppressed errors.

Reply:
Imo the fact that this isn't a constant property is just an implementation detail, and it should be named as if it was one.